test/e2e/testdata/artifactgc/artgc-template-2.yaml

```diff
@@ -0,0 +1,52 @@
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: artgc-template-2
+spec:
+  artifactGC:
+    strategy: OnWorkflowDeletion 
+  workflowMetadata:
```

Comment:
unlike the WorkflowTemplate test that was already in there, this one includes a strategy specified on the Workflow level